sdk/core/core-rest-pipeline/src/retryStrategies/exponentialRetryStrategy.ts

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { PipelineResponse } from "../interfaces";
+import { RestError } from "../restError";
+import { getRandomIntegerInclusive } from "../util/helpers";
+import { RetryStrategy } from "./retryStrategy";
+import { isThrottlingRetryResponse } from "./throttlingRetryStrategy";
+
+// intervals are in milliseconds
+const DEFAULT_CLIENT_RETRY_INTERVAL = 1000;
+const DEFAULT_CLIENT_MAX_RETRY_INTERVAL = 1000 * 64;
+
+/**
+ * A retry strategy that retries with an exponentially increasing delay in these two cases:
+ * - When there are errors in the underlying transport layer (e.g. DNS lookup failures).
+ * - Or otherwise if the outgoing request fails (408, greater or equal than 500, except for 501 and 505).
+ */
+export function exponentialRetryStrategy(
+  options: {
+    /**
+     * The amount of delay in milliseconds between retry attempts. Defaults to 1000
+     * (1 second.) The delay increases exponentially with each retry up to a maximum
+     * specified by maxRetryDelayInMs.
+     */
+    retryDelayInMs?: number;
+
+    /**
+     * The maximum delay in milliseconds allowed before retrying an operation. Defaults
+     * to 64000 (64 seconds).
+     */
+    maxRetryDelayInMs?: number;
+
+    /**
+     * If true it won't retry if it received a system error.
+     */
+    ignoreSystemErrors?: boolean;
+
+    /**
+     * If true it won't retry if it received a non-fatal HTTP status code.
+     */
+    ignoreHttpStatusCodes?: boolean;
+  } = {}
+): RetryStrategy {
+  const retryInterval = options.retryDelayInMs ?? DEFAULT_CLIENT_RETRY_INTERVAL;
+  const maxRetryInterval = options.maxRetryDelayInMs ?? DEFAULT_CLIENT_MAX_RETRY_INTERVAL;
+
+  let retryAfterInMs = retryInterval;
+
+  return {
+    name: "exponentialRetryStrategy",
+    retry({ retryCount, response, responseError }) {
+      if (isThrottlingRetryResponse(response)) {
+        return { skipStrategy: true };
+      }
+      const skippingIfSystemError = !options.ignoreSystemErrors && !isSystemError(responseError);
+      const skippingIfHttpStatusCode =
+        !options.ignoreHttpStatusCodes && !isExponentialRetryResponse(response);
+      if (skippingIfSystemError || skippingIfHttpStatusCode) {
+        return { skipStrategy: true };
+      }
+
+      // Exponentially increase the delay each time
+      const exponentialDelay = retryAfterInMs * Math.pow(2, retryCount);
+      // Don't let the delay exceed the maximum
+      const clampedExponentialDelay = Math.min(maxRetryInterval, exponentialDelay);
+      // Allow the final value to have some "jitter" (within 50% of the delay size) so
+      // that retries across multiple clients don't occur simultaneously.
+      const delayWithJitter =
+        clampedExponentialDelay / 2 + getRandomIntegerInclusive(0, clampedExponentialDelay / 2);
+
+      retryAfterInMs = delayWithJitter;
+      return { retryAfterInMs };
```

Comment:
nitpick: remove the local variable `retryAfterInMs`?

Reply:
I’m re-using the local `retryAfterInMs` to keep changing its value, since the `information` object does not contain a `retryAfterInMs` property. But I can clean this up this way: https://github.com/Azure/azure-sdk-for-js/pull/19078/commits/f3f36949964707e1325f3ad21b03486200c36595